### PR TITLE
feature: add support for native xDAI in Curve

### PR DESCRIPTION
**Pull request by adamazad**
# Summary

Gnosis Chain's native asset is the xDAI stablecoin. This PR adds support for trading between xDAI and USDC and xDAI and USDT by using a un/wrapper contract that handles deposits and withdrawals.

### CODE REVIEW

File: src/entities/trades/curve/CurveTrade.ts [hunk 1]
```diff
@@ -40,6 +42,8 @@ interface QuoteFromPool {
   error?: Error
 }
 
+const debugCurveGetQuote = debug('ecoRouter:curve:getQuote')
```
Comment (jorgedclopes) [inline src/entities/trades/curve/CurveTrade.ts:45]: can you elaborate a bit on what debug is and why we need it in develop?

File: src/entities/trades/curve/CurveTrade.ts [hunk 2]
```diff
@@ -482,11 +544,15 @@ export class CurveTrade extends Trade {
     invariant(chainId !== undefined && RoutablePlatform.CURVE.supportsChain(chainId), 'CHAIN_ID')
 
     try {
+      // Custom handler for xDAI<>USDT and xDAI<>USDC
+      // on Gnosis Chain
```
Comment (jorgedclopes) [inline src/entities/trades/curve/CurveTrade.ts:548]: remove before merging or elaborate on docstring

File: src/entities/trades/curve/CurveTrade.ts [hunk 3]
```diff
@@ -220,26 +297,16 @@ export class CurveTrade extends Trade {
...
-
-    // Exchange fee
-    const exchangeRateWithoutFee = 1
-    const exchangeRate = 1 - FEE_DECIMAL
-
     // If a pool is found
     // Ignore the manual off-chain search
     if (bestPoolAndOutputRes) {
-      debug(`Curve::GetQuote | Found best pool from Curve`, bestPoolAndOutputRes)
+      debugCurveGetQuote(`Found best pool from Curve registery`, bestPoolAndOutputRes)
```
Comment (Diogomartf) [inline src/entities/trades/curve/CurveTrade.ts:303]: I think there is a typo

```suggestion
      debugCurveGetQuote(`Found best pool from Curve registry`, bestPoolAndOutputRes)
```

File: src/entities/trades/curve/types.ts [hunk 4]
```diff
@@ -36,14 +37,16 @@ export interface CurveTradeQuote {
   contract: Contract
 }
 
-export interface CurveTradeBestTradeExactInParams {
+interface CurveGetTradeCommonParams {
+  maximumSlippage: Percent
+  receiver?: string
+}
+export interface CurveTradeBestTradeExactInParams extends CurveGetTradeCommonParams {
```
Comment (jorgedclopes) [inline src/entities/trades/curve/types.ts:44]: There are lines separating other objects. It's not functionally important but let's lint the code.

**Review by jorgedclopes — APPROVED** [review]
I trust you to address the comments.
I can't possibly review if the ABI is good, I skipped it in my review.
Also, I don't know much about the feature that is implemented here so my "limited" review was mainly focused on code quality.
Try to elaborate the description a bit further in the future so people not so close to the work you're doing can get more context. Specially on how we can confirm this does what we need, whatever that may be.
Good coding.

File: src/entities/trades/curve/contracts/contracts.ts [hunk 5]
```diff
@@ -118,3 +119,11 @@ export async function getExchangeRoutingInfo({
   }
   return
 }
+
+/**
+ * Returns an instance of the Curve DAI Exchange contract.
+ * This contract is only available on Gnosis Chain.
+ */
```
[the remark this answers is not in the document]
Comment (adamazad) [reply src/entities/trades/curve/contracts/contracts.ts:126]: Do you mean JSDoc?

File: src/entities/trades/curve/types.ts (hunk 4 above)
Comment (adamazad) [reply src/entities/trades/curve/types.ts]: Addressed in https://github.com/levelkdev/swapr-sdk/pull/80/commits/55595d108ce509867f961b739bfea556f9cc792f

File: src/entities/trades/curve/CurveTrade.ts (hunk 3 above)
Comment (adamazad) [reply src/entities/trades/curve/CurveTrade.ts]: Addressed in https://github.com/levelkdev/swapr-sdk/pull/80/commits/55595d108ce509867f961b739bfea556f9cc792f

File: src/entities/trades/curve/CurveTrade.ts (hunk 2 above)
Comment (adamazad) [reply src/entities/trades/curve/CurveTrade.ts]: Addressed in https://github.com/levelkdev/swapr-sdk/pull/80/commits/55595d108ce509867f961b739bfea556f9cc792f

File: src/entities/trades/curve/contracts/contracts.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (adamazad) [reply src/entities/trades/curve/contracts/contracts.ts]: I addressed this in https://github.com/levelkdev/swapr-sdk/pull/80/commits/55595d108ce509867f961b739bfea556f9cc792f, but feel free to open a PR later on

**Pull request merged**